pkg/controller/datavolume-controller.go

```diff
@@ -2779,6 +2787,35 @@ func getDeltaTTL(dv *cdiv1.DataVolume, ttl int32) time.Duration {
 	return delta
 }
 
+// handleCloneWithoutSource does error handling and validation of clones without source PVC
+func (r *DatavolumeReconciler) handleCloneWithoutSource(datavolume *cdiv1.DataVolume) error {
+	sourcePvc, err := r.findSourcePvc(datavolume)
+	if err != nil {
+		// Clone without source
+		if k8serrors.IsNotFound(err) {
+			datavolume.Annotations[AnnNoSourceClone] = "true"
+			r.updateDataVolume(datavolume)
+		}
+		return err
+	}
+
+	noSource, available := datavolume.Annotations[AnnNoSourceClone]
+	if available && noSource == "true" {
+		// Since the admission webhook couldn't validate the clone against its source PVC
+		// when creating the DV, we do the validation now
+		err := ValidateClone(sourcePvc, &datavolume.Spec)
+		if err != nil {
+			return err
+		}
+
+		// The clone has a source now
+		datavolume.Annotations[AnnNoSourceClone] = ""
```

Comment:
So the annotation is removed to do it only once? comment will be nice

Reply:
Exactly, since the webhook validation is skipped when creating the DV, I added that annotation to make sure we still check it once the source PVC is created. However, I don't see the point of keeping the annotation.